src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1839,6 +1841,12 @@ protected function prepareBaseUrl()
             return $prefix;
         }
 
+        if ($this->isFromTrustedProxy() && $trustedPrefix = $this->getTrustedValues(self::HEADER_X_FORWARDED_PREFIX)) {
+            $trustedPrefix = rtrim($trustedPrefix[0], '/');
+
+            return $trustedPrefix;
```

Comment:
AFAICT, the proxy prefix must be prepended to any prefix being needed at the webserver level, rather than replacing it entirely. Otherwise, this breaks things when the webserver itself expects a prefix too.

Reply:
Good point! This actually turned out to be a little more complex. We actually need both values baseURL with and without proxy prefix to make everything work fine for custom scenarios like deploying code in a subdir. 

I have added a test for that and adjusted code accordingly. Please have a look!